oshdb-util/src/main/java/org/heigit/ohsome/oshdb/util/celliterator/CellIterator.java

```diff
@@ -802,6 +804,22 @@ public IterateAllEntry(
     });
   }
 
-}
-
-
+  private static SortedMap<OSHDBTimestamp, OSMEntity> getVersionsByTimestamps(
+      OSHEntity osh, List<OSHDBTimestamp> byTimestamps) {
+    SortedMap<OSHDBTimestamp, OSMEntity> result = new TreeMap<>();
+
+    int i = byTimestamps.size() - 1;
+    Iterator<? extends OSMEntity> itr = osh.getVersions().iterator();
```

Comment:
we should check if we need to have a map returned instead of an sorted list. The versions from `osh.getVersions()` are already sorted by version_number in decreasing, maybe we can use this fact.
I checked the call sites for getVersionsbyTimestamp and in the current usage, it will only be iterated over the map.
So returning here a list instead of a map could be better performance wise.

In the moment, because `osh.getVersions()` returns timestamps in  decreasing, but the TreeMap sorts the timestamps in increasing order, I think, there are far more checks then using a list, and sort afterwards (and remove duplicates )

Reply:
Good suggestion!

A quick test revealed that a simple `Lists.reverse()` is sufficient (at least to make the unit tests pass), so perhaps one doesn't even need to actually sort and/or deduplicate the result.

But I would be slightly hesitant to include this change in this PR, since the goal was not to do any major changes or performance improvements. It was just to make the code better organized, cleaned up and better documented. I would however appreciate it if you could create a separate issue for this suggestion, than we can work on it after this PR is merged.

Reply:
Good point, I will then open a new issue/pr for improving the way we generate the contribution. 👍

Reply:
Merci. Btw: feel free to open an issue already now, then you don't have to keep it in mind for too long. :wink: 

Also, I have already an almost complete implementation of your suggestion in my `git stash`, in case you want to skip the boring implementation details.

Reply:
Cool, thank you looking forward to work together with you!